qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -77,13 +99,14 @@ def __init__(
             self.decomposer = TwoQubitBasisDecomposer(kak_basis_gate)
         elif basis_gates is not None:
             kak_gates = KAK_GATE_NAMES.keys() & (basis_gates or [])
+            kak_param_gates = KAK_GATE_PARAM_NAMES.keys() & (basis_gates or [])
             if kak_gates:
                 self.decomposer = TwoQubitBasisDecomposer(
                     KAK_GATE_NAMES[kak_gates.pop()], basis_fidelity=approximation_degree or 1.0
                 )
-            elif "rzx" in basis_gates:
-                self.decomposer = TwoQubitBasisDecomposer(
-                    CXGate(), basis_fidelity=approximation_degree or 1.0
+            elif kak_param_gates:
+                self.decomposer = TwoQubitControlledUDecomposer(
+                    KAK_GATE_PARAM_NAMES[kak_param_gates.pop()]
```

Comment:
I remember a discussion we had in unitary synthesis where we agreed not to use `pop` because of result reproducibility. I think the same logic would apply in this case.

Reply:
done in [9095358](https://github.com/Qiskit/qiskit/pull/13884/commits/9095358f5f771e51d4e2013dca9b9f6d947eea95)